mava/adders/reverb/base.py

```diff
@@ -115,6 +154,214 @@ def __init__(
+                            for key in trajectory.extras.keys():
+                                if (
+                                    type(trajectory.extras[key]) is dict
+                                    and cur_agent in trajectory.extras[key]
+                                ):
+                                    new_trajectory.extras[key][
+                                        want_agent
+                                    ] = trajectory.extras[key][cur_agent]
+                                else:
+                                    # TODO: (dries) Only actually need to do this once
+                                    # and not per agent. Maybe fix this in the future.
+                                    new_trajectory.extras[key] = trajectory.extras[key]
+                                if type(trajectory) == mava_types.Transition:
+                                    ext = trajectory.next_extras[key]  # type: ignore
+                                    if (
+                                        type(ext) is dict  # type: ignore
+                                        and cur_agent in ext  # type: ignore
+                                    ):
+                                        new_trajectory.next_extras[key][  # type: ignore
+                                            want_agent
+                                        ] = trajectory.next_extras[  # type: ignore
+                                            key
+                                        ][  # type: ignore
+                                            cur_agent
+                                        ]  # type: ignore
+                                    else:
+                                        # TODO: (dries) Only actually need to
+                                        # do this once and not per agent. Maybe
+                                        # fix this in the future.
+                                        new_trajectory.next_extras[  # type: ignore
+                                            key
+                                        ] = trajectory.next_extras[  # type: ignore
+                                            key
+                                        ]  # type: ignore
+
+                        # Write the new_trajectory to the table.
+                        self._writer.create_item(
+                            table=table, priority=priority, trajectory=new_trajectory
+                        )
+            if not created_item:
+                raise EOFError(
+                    "This experience was not used by any trainer: ",
+                    trajectory.actions.keys(),
+                )
+        else:
+            # Default setting (deprecate this) with only one table. In this setting
+            # we write the entire trajectory to that table.
+            table_items = list(table_priorities.items())
+            assert len(table_items) == 1
```

Comment:
Why are we assuming there is one table here? Shouldn't we just ` for table_name, priority in table_priorities.items():` ...  & write, e.g. for systems that haven't implemented multiple trainers yet?

Reply:
I am replying to the above comment here. For the moment it has to stay like that yes. The vmin and vmax cannot be kwargs. But if you can think about a cleaner way to do this let me know? Maybe having a dictionary with kwargs in instead of a if statement?

For the second comment. Maybe I am mistaken, but I don't think any system uses more than one table except for multiple trainers. But I am happy to change it back if you think it allows for more flexibility. Let my know.

Reply:
1.) I like where the solution is going!  made a slight adjustment to the code, but the idea is there. Let me know if you are happy.
2.) We won't ignore tables due to the assert. So if there is more than one table the user will know about it. But sure I don't mind adding it back. I fixed it.